### PR TITLE
Jigglypuff tweaks and turnaround bair removals

### LINKED ISSUE #1312: jigglypuff bair turnaround removal and various tweaks

### PR DESCRIPTION
**Pull request by TeamPearson**
Started on removing bair turnaround and ended up rebalancing various aspects of puff

fixes #1312 

Rework back air to incorporate no turn around. Evaluated frame data and hitbox to accompany this new trait.✅\

Rest is active for 3 frames and is nearly double the size it has been in the past and comes out of frame 3 as well. This feels like an intended nerf but ends up having more leniency than before. PLAN: Reduce hitbox size, active frames, and make the hitbox come out frame 1. ✅

No throws are in ACMD. View throws and evaluate whether or not they need changes perhaps up throw needs a buff.✅
Up air. Buff damage stat and rework the kb to either keep the move in line or make it behave better. Lower active frames from 13 to 4 lmao. Shave 2 frames off of FAF.✅

Side special is much  more disjointed than it should be and 2 frames more active than it should be this should be corrected by moving the move further into puffs body and reducing size.✅

Restore jigglypuff shield break instadeath✅

Clean up effects✅

Add back nair autocancel in a shorthop✅

Nerf rest on hit cancel window by 40 frames✅

hitbox comparison and in depth changelog will be added soon

New Bair animation:  Credit to Bottz3 for creating this animation for me.

[fighter.zip](https://github.com/HDR-Development/HewDraw-Remix/files/10440249/fighter.zip)

### CODE REVIEW

**Comment by hdr-balance-backroom** [comment]
Fix the rest bug

File: fighters/purin/src/acmd/aerials.rs [hunk 1]
```diff
@@ -110,23 +116,41 @@ unsafe fn purin_attack_air_hi_game(fighter: &mut L2CAgentBase) {
...
+        FT_MOTION_RATE(fighter, 0.2);
+    if is_excute(fighter) {
+        EFFECT_FOLLOW_FLIP_ALPHA(fighter, Hash40::new("sys_attack_arc"), Hash40::new("sys_attack_arc"), Hash40::new("top"), 0, 8, 1, 0, -110, -90, 0.7, true, *EF_FLIP_YZ, 0.3);
+        LAST_EFFECT_SET_RATE(fighter, 1.2);
+    frame(lua_state, 22.0);
+    if is_excute(fighter) {
+        FT_MOTION_RATE(fighter, 1.5)
+    }
+    }
+}
```
Comment (SuddyN) [inline fighters/purin/src/acmd/aerials.rs]: - Shouldn't use `FT_MOTION_RATE` calls in effect ACMD (duplicated in this move's game ACMD)
  - `FT_MOTION_RATE(fighter, 0.2)` on line 143 (on frame 5) is actually `FT_MOTION_RATE(fighter, 3.5/(22.0-9.0))` in game ACMD (on frame 9)
- `if is_execute()` block on line 144 is not closed until line 151 - should be closed before line 147

```suggestion
#[acmd_script( agent = "purin", script = "effect_attackairhi", category = ACMD_EFFECT, low_priority )]
unsafe fn purin_effect_attackairhi(fighter: &mut L2CAgentBase) {
    let lua_state = fighter.lua_state_agent;
    let boma = fighter.boma();
    frame(lua_state, 5.0);
    if is_excute(fighter) {
        EFFECT_FOLLOW_FLIP_ALPHA(fighter, Hash40::new("sys_attack_arc"), Hash40::new("sys_attack_arc"), Hash40::new("top"), 0, 8, 1, 0, -110, -90, 0.7, true, *EF_FLIP_YZ, 0.3);
        LAST_EFFECT_SET_RATE(fighter, 1.2);
    }
}
```

File: fighters/purin/src/acmd/throws.rs [hunk 2]
```diff
@@ -2,8 +2,95 @@
...
 
 
-pub fn install() {
+#[acmd_script( agent = "purin", script = "game_throwb" , category = ACMD_GAME , low_priority)]
+unsafe fn game_throwb(fighter: &mut L2CAgentBase) {
+    let lua_state = fighter.lua_state_agent;
+    let boma = fighter.boma();
+        if is_excute(fighter) {
+            ATTACK_ABS(fighter, *FIGHTER_ATTACK_ABSOLUTE_KIND_THROW, 0, 10.0, 135, 25, 0, 90, 0.0, 1.0, *ATTACK_LR_CHECK_F, 0.0, true, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_NONE, *ATTACK_REGION_THROW);
+            ATTACK_ABS(fighter, *FIGHTER_ATTACK_ABSOLUTE_KIND_CATCH, 0, 10.0, 361, 100, 0, 40, 0.0, 1.0, *ATTACK_LR_CHECK_F, 0.0, true, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_NONE, *ATTACK_REGION_THROW);
```
Comment (SuddyN) [inline fighters/purin/src/acmd/throws.rs:11]: If I'm reading this correctly, BThrow will now do 20%?

``` suggestion
            ATTACK_ABS(fighter, *FIGHTER_ATTACK_ABSOLUTE_KIND_THROW, 0, 10.0, 135, 25, 0, 90, 0.0, 1.0, *ATTACK_LR_CHECK_F, 0.0, true, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_NONE, *ATTACK_REGION_THROW);
            ATTACK_ABS(fighter, *FIGHTER_ATTACK_ABSOLUTE_KIND_CATCH, 0, 3.0, 361, 100, 0, 40, 0.0, 1.0, *ATTACK_LR_CHECK_F, 0.0, true, Hash40::new("collision_attr_normal"), *ATTACK_SOUND_LEVEL_S, *COLLISION_SOUND_ATTR_NONE, *ATTACK_REGION_THROW);
```

**Review by SuddyN — CHANGES_REQUESTED** [review]

File: fighters/purin/src/acmd/throws.rs (hunk 2 above)
Comment (TeamPearson) [reply fighters/purin/src/acmd/throws.rs:11]: It does not do 20 percent

File: fighters/purin/src/acmd/aerials.rs (hunk 1 above)
Comment (TeamPearson) [reply fighters/purin/src/acmd/aerials.rs]: new commit should fix this stuff

**Review by SuddyN — DISMISSED** [review]
LGTM

File: fighters/purin/src/acmd/aerials.rs [hunk 3]
```diff
@@ -24,13 +24,8 @@ unsafe fn purin_attack_air_n_game(fighter: &mut L2CAgentBase) {
     if is_excute(fighter) {
         FT_MOTION_RATE(fighter, 1.125);
         AttackModule::clear_all(boma);
-    }
```
Comment (jobrien97) [inline fighters/purin/src/acmd/aerials.rs:27]: Not sure why this change was made to nair, it's not listed in the changelog. This makes nair autocancel as soon as its hitboxes clear and extends its endlag

**Review by hdr-balance-backroom — APPROVED** [review]
Lgtm
